hydromt_wflow/components/config.py

```diff
@@ -1,43 +1,131 @@
+from tomllib import load as load_toml
 
 from hydromt_wflow.utils import get_config, set_config
 
+__all__ = ["WflowConfigComponent"]
 
-class WflowConfigComponent(ConfigComponent):
-    """A class to manage the configuration of wflow models."""
+logger = logging.getLogger(f"hydromt.{__name__}")
+
+
+class WflowConfigComponent(ModelComponent):
+    """Manage the wflow configurations.
+
+    Parameters
+    ----------
+    model : Model
+        HydroMT model instance
+    filename : str
+        A path relative to the root where the configuration file will
+        be read and written if user does not provide a path themselves.
+        By default 'wflow_sbm.toml'
+    default_template_filename : str, optional
+        A path to a template file that will be used as default in the ``create``
+        method to initialize the configuration file if the user does not provide
+        their own template file. This can be used by model plugins to provide a
+        default configuration template. By default None.
+    """
 
     def __init__(
         self,
         model: Model,
         *,
         filename="wflow_sbm.toml",
-        default_template_filename: str | None = None,
     ):
-        super().__init__(
-            model=model,
-            filename=filename,
-            default_template_filename=default_template_filename,
-        )
+        self._data: tomlkit.TOMLDocument[str, Any] | None = None
+        self._filename: str = filename
 
-    def set(self, *args):
-        """Set the config options."""
-        self._initialize()
-        set_config(self._data, *args)
+        super().__init__(model=model)
 
-    def get(self, *args) -> Any | None:
-        """Get config options."""
-        self._initialize()
-        return get_config(self._data, *args)
+    def __eq__(self, other: ModelComponent):
+        """Compare components based on content."""
+        other_config = cast(WflowConfigComponent, other)
+        return self.data == other_config.data
 
+    ## Private
     def _initialize(self, skip_read=False) -> None:
         """Initialize the model config."""
         if self._data is None:
             self._data = tomlkit.TOMLDocument()
             if self.root.is_reading_mode() and not skip_read:
                 self.read()
+
+    ## Properties
+    @property
+    def data(self) -> tomlkit.TOMLDocument[str, Any]:
+        """Model config values."""
+        if self._data is None:
+            self._initialize()
+        return self._data
+
+    ## I/O Methods
+    @hydromt_step
+    def read(
+        self,
+        path: Path | str | None = None,
+    ):
+        """Read the wflow configurations file."""
+        self._initialize(skip_read=True)
+
+        # Solve pathing
+        p = path or self._filename
+        read_path = Path(self.root.path, p)
+
+        # Check if the file exists
+        if read_path.is_file():
+            logger.info(f"Reading model config file from {read_path.as_posix()}.")
```

Comment:
I think we should follow the system convention that we are running on so I'd remove the `as_posix()` (even though I also prefer it)